### DISCUSSION TITLE
Round Corners!!!

### DISCUSSION THREAD

**Discussion by Jackson-Nickk** [linuxmint/discussions#500] [Ideas]
I don't know if this idea is duplicated but I couldn't find any other discussion about it, which is weird...

Make all corners of the windows and pop ups round. I don't know why this scheme is common in old designs. that a window has rounded title bar corners but not bottom ones. GTK 2 and 3 has that, KDE Plasma has it by default but it can be changed in the settings.

I don't get it. If a round corners are possible to implement, why it is implemented half way?
I know that Cinnamon desktop is based on GTK 2 and that's why it follows old implementations, but it should not be an absolute limitation and prevent new designs right? isn't that one of the goals for Cinnamon to be extended where GTK 2 died?

Sorry for my newbiness and lack of background information on the subject. I just want Cinnamon to be better.

**Comment by fredcw** [comment]
>GTK 2 and 3 has that

Actually, gtk2 & 3 can't have rounded bottom corners for application windows, and that's the issue. Gtk 4 does but not gtk2 & 3.

**Reply by Jackson-Nickk** [reply]
I know that it's not possible. What confuses me is why? If title bars can be round, then what prevents it to apply to bottom ones too?

**Reply by fredcw** [reply]
With gtk2 & 3 (and other toolkits), the window borders and title bar (called window decorations) are not drawn or handled by the app's toolkit (gtk2 or 3 in this case). Instead they are drawn and handled by the window manager (muffin). The titlebar can be rounded because doing so does not encroach on the app's allocated space whereas rounding the bottom corners would encroach on the rectangular area allocated to the application. I suppose the window manager theming language (metacity) could have allowed for extra wide side and bottom borders to allow room for rounded corners but it just wasn't designed that way and it's too late to change it now without breaking backward compatibility with all the metacity themes already out there. Also, the extra thick borders probably wouldn't look too good. Gtk4 is able to draw rounded corners because it draws the window decorations itself and, called client side decorations (CSD), and has allowed for this in it's .css stylesheets.

**Reply by Jackson-Nickk** [reply]
Good points. But for how long will Cinnamon be dependent on an obsolete framework? Are we gonna be GTK 2 until eternity?

**Reply by Qronikarz** [reply]
I think Mint is primarily GTK3. At least according to the comments on the blog.
And yes, currently the plan is to stay on GTK3 for unspecified time.
https://blog.linuxmint.com/?p=4675

Maybe in the future they will also follow Budgie and Cosmic and move away from GTK, but this is Mint so it will take time.

**Comment by mfreeman72** [comment]
This issue has been discussed quite a lot over the years, actually. The newest discussions that I've been involved with are here:
https://github.com/linuxmint/cinnamon/issues/7530
https://github.com/linuxmint/muffin/issues/652

**Comment by carsten88** [comment]
I vote this up!!!

I'm working on a theme and came to the uppleasant surprise the bottom corners can't be styled. 
see this thread https://forums.linuxmint.com/viewtopic.php?t=444116

I'm don't know about how gtk works and all. But all that seems to be needed for this is a stylable element at the bottom of the window. Maybe that can be included in the next version of Mint with (or without) gtk3?

**Comment by JosephMcc** [comment]
> I'm don't know about how gtk works and all. But all that seems to be needed for this is a stylable element at the bottom of the window.

Upvoting isn't going to change this. @fredcw did a pretty good job of explaining up above why this doesn't work. His comments about Gtk4 aren't quite true though. Gtk4 apps don't "just support" bottom rounded corners. Client side decorated apps do. Server side decorated apps do not because how the window and window frame are created as two separate objects. It's not about the toolkit.

**Reply by Jopp-gh** [reply]
Since "rounded corners" comes up quite often: if possible, somebody should pin this thread to be visible all times.
